src/startup/ModeCLIParser.cpp

```diff
@@ -0,0 +1,158 @@
+/**
+ * Copyright (c) 2019 DKFZ - ODCF
+ *
+ * Distributed under the MIT License (license terms are at https://github.com/dkfz-odcf/FastqIndEx/blob/master/LICENSE.txt).
+ */
+
+#include "process/io/PathSink.h"
+#include "process/io/PathSource.h"
+#include "process/io/s3/S3Sink.h"
+#include "process/io/s3/S3Source.h"
+#include "process/io/Source.h"
+#include "process/io/ConsoleSink.h"
+#include "process/io/StreamSource.h"
+#include "ModeCLIParser.h"
+#include <tclap/CmdLine.h>
+
+#include <memory>
+
+_IntValueArg ModeCLIParser::createVerbosityArg(CmdLine *cmdLineParser) const {
+    return _makeIntValueArg(
+            "v",
+            "verbosity",
+            string("Sets the verbosityArg of the application in the range of 0 (default, less) to 3 (debug, max). ") +
+            "Invalid values will be ignored and the default of 0 will apply. -D automatically sets the level to 3.",
+            false,
+            0, cmdLineParser);
+}
+
+_SwitchArg ModeCLIParser::createDebugSwitchArg(CmdLine *cmdLineParser) const {
+    return _makeSwitchArg(
+            "D", "enableDebugging",
+            string("Only practicable when you debug the application, e.g. with an IDE. This will tell the extractor ") +
+            "component to store various debug information during runtime.",
+            cmdLineParser);
+}
+
+_StringValueArg ModeCLIParser::createIndexFileArg(CmdLine *cmdLineParser) const {
+    return _makeStringValueArg(
+            "i", "indexFile",
+            string("The index file which shall be created or - for stdout or \"\" to append .fqi to the FASTQ filename. ") +
+            "Note, that the index will be streamed to stdout, if you provide \"\" or - as the FASTQ file parameter. " +
+            "The index file can also reside in an S3 bucket. Enter the filename here like s3:<filename> and set the"
+            " bucket with --bucket." +
+            " Note, that this software will create the index in the tmp directory before uploading it to S3!",
+            false, "",
+            cmdLineParser);
+}
+
+_StringValueArg ModeCLIParser::createFastqFileArg(CmdLine *cmdLineParser) const {
+    return _makeStringValueArg(
+            "f", "fastqFile",
+            string("The FASTQ file which shall be indexed or - for stdin.") +
```

Comment:
```suggestion
            string("The FASTQ file which shall be indexed or \"-\" for stdin.") +
```

I '-' looks like are dash. Please always quote.